src/pocketmine/level/Level.php

```diff
@@ -782,6 +805,42 @@ public function doTick(int $currentTick){
 			$this->checkSleep();
 		}
 
+		if(!$this->weatherLocked){
+			--$this->thunderTime;
+			--$this->rainTime;
+			--$this->clearTime;
+			if($this->weather == self::WEATHER_CLEAR and ($this->rainTime <= 0 or $this->thunderTime <= 0)){
```

Comment:
Sorry to be annoying...but you have a mixture of `==` and `===`. Do you mind changing them to the latter if possible?

Othee than that, LGTM